### PR TITLE
Mutate public to private methods

### PR DESCRIPTION
**Pull request by deleugpn**
This PR:

- [x] Adds new feature ...
- [x] Covered by tests
- [x] Doc PR https://github.com/infection/site/pull/68

Fixes https://github.com/infection/infection/issues/498

This PR adds the ability to mutate from `public` straight to `private` methods.

### CODE REVIEW

**Comment by sanmai** [comment]
1. CI is failing because you need to run E2E tests and update fixtures for them (as now they see more mutations). The latter is typically done with `./tests/e2e_tests`.
2. I'd make a `private function makeClassMethod($visiblity): ClassMethod` and avoid copy'n'pasting a bunch of code, what do you think?

**Comment by deleugpn** [comment]
Sorry, I did this PR at GitHub opening event in Amsterdam and forgot to check back the status. I'll look into the e2e test.
I swear I tried making a private method as you said and it was breaking the mutation.
I'll try again =D

**Comment by sanmai** [comment]
Please do. I'm here if you need a hand.

File: src/Mutator/FunctionSignature/PublicVisibility.php [hunk 1]
```diff
@@ -24,16 +24,16 @@
...
      * @param Node $node
      *
-     * @return ClassMethod
+     * @return \Generator
      */
     public function mutate(Node $node)
     {
         /* @var ClassMethod $node */
-        return new ClassMethod(
+        yield new ClassMethod(
```
Comment (miguelgf) [inline src/Mutator/FunctionSignature/PublicVisibility.php]: What improvement adds the yield here instead of the return?

Comment (borNfreee) [reply src/Mutator/FunctionSignature/PublicVisibility.php]: this is a new feature (not yet released). It allows to return `N` mutations from one mutator. Note the second mutation on line 48 below.

Also, this is not the same as `reutnr [new ClassMethod(...), new ClassMethod(...)]`. Such code replaces 1 `Node` with an array of `Node`s, which is completely different behaviour.

**Comment by borNfreee** [comment]
Are you going to finish this PR @deleugpn or can we take over it?

**Comment by deleugpn** [comment]
Sorry, you can take it over. I hope to be back at some point.

**Pull request closed without merging**